src/VisualStudio/Core/Def/Implementation/LanguageClient/InProcLanguageServer.cs

```diff
@@ -0,0 +1,232 @@
+        public async Task<object[]> GetTextDocumentReferencesAsync(JToken input, CancellationToken cancellationToken)
+        {
+            var referenceParams = input.ToObject<ReferenceParams>();
+            await _threadingContext.JoinableTaskFactory.SwitchToMainThreadAsync(cancellationToken);
+            return await this._protocol.FindReferencesAsync(_workspace.CurrentSolution, referenceParams, _clientCapabilities, cancellationToken).ConfigureAwait(false);
+        }
+
+        [JsonRpcMethod(Methods.TextDocumentSignatureHelpName)]
+        public async Task<SignatureHelp> GetTextDocumentSignatureHelpAsync(JToken input, CancellationToken cancellationToken)
+        {
+            var textDocumentPositionParams = input.ToObject<TextDocumentPositionParams>();
+            return await this._protocol.GetSignatureHelpAsync(_workspace.CurrentSolution, textDocumentPositionParams, _clientCapabilities, cancellationToken).ConfigureAwait(false);
+        }
+
+        [JsonRpcMethod(Methods.WorkspaceSymbolName)]
+        public async Task<SymbolInformation[]> GetWorkspaceSymbolsAsync(JToken input, CancellationToken cancellationToken)
+        {
+            var workspaceSymbolParams = input.ToObject<WorkspaceSymbolParams>();
+            return await this._protocol.GetWorkspaceSymbolsAsync(_workspace.CurrentSolution, workspaceSymbolParams, _clientCapabilities, cancellationToken).ConfigureAwait(false);
+        }
+
+#pragma warning disable VSTHRD100 // Avoid async void methods
+        private async void DiagnosticService_DiagnosticsUpdated(object sender, DiagnosticsUpdatedArgs e)
+#pragma warning restore VSTHRD100 // Avoid async void methods
+        {
+            // Since this is an async void method, exceptions here will crash the host VS. We catch exceptions here to make sure that we don't crash the host since
+            // the worst outcome here is that guests may not see all diagnostics.
+            try
+            {
+                // LSP doesnt support diagnostics without a document. So if we get project level diagnostics without a document, ignore them.
+                if (e.DocumentId != null && e.Solution != null)
+                {
+                    var document = e.Solution.GetDocument(e.DocumentId);
+                    if (document == null || document.FilePath == null)
+                    {
+                        return;
+                    }
+
+                    // Only publish document diagnostics for the languages this provider supports.
+                    if (document.Project.Language != LanguageNames.CSharp && document.Project.Language != LanguageNames.VisualBasic)
+                    {
+                        return;
+                    }
+
+                    // LSP does not currently support publishing diagnostics incrememntally, so we re-publish all diagnostics.
+                    var diagnostics = await GetDiagnosticsAsync(e.Solution, document, CancellationToken.None).ConfigureAwait(false);
```

Comment:
cancellationToken.none is concerning as well.

Reply:
See previous and https://github.com/dotnet/roslyn/issues/40388